### PR TITLE
Rework synchronous exports

### PR DESCRIPTION
**Pull request by jnm**
* Regenerate the same export at most once every 5 minutes (configurable in the Django admin interface)
* Move synchronous exports to `data.csv` and `data.xlsx` nested underneath the export settings detail view
* Remove `ExportObjectMixin` in favor of a base class from which a new `SynchronousExport` model inherits
* Set a reasonable execution time limit on MongoDB queries initiated by (all) synchronous requests while continuing to allow a higher limit for asynchronous (Celery) queries
* Fix some warnings related to time zones

## Checklist

1. [x] If you've added code that should be tested, add tests
2. [x] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a description of your work suitable for publishing on [our forum](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

Synchronous CSV and XLSX exports are now available at `/api/v2/assets/<code>{asset_uid}</code>/export-settings/<code>{setting_uid}</code>/data.csv` (and `…/data.xlsx`). If an export is requested multiple times for the same settings, it will be regenerated only once every 5 minutes. Synchronous exports may fail for large projects, as their processing must complete within the web application server's 2-minute timeout (compared with 30 minutes for asynchronous export tasks).

This is an API-only feature.

## Related issues

Related to #3700, #3586, #3319. Supersedes #3714.

### CODE REVIEW

**Review by joshuaberetta — CHANGES_REQUESTED** [review]
@jnm I'm seeing this interesting behaviour with the caching:

```
>>> # fetch data using Google Sheets
>>> SynchronousExport.objects.filter(user=u)
<QuerySet []>
>>> # make a request outside of the browser using wget
>>> SynchronousExport.objects.filter(user=u)
<QuerySet []>
>>> # make a request with the browser
>>> SynchronousExport.objects.filter(user=u)
<QuerySet [<SynchronousExport: SynchronousExport object (17)>]>
```

File: kpi/models/import_export_task.py [hunk 1]
```diff
@@ -614,10 +800,84 @@ def remove_excess(cls, user, source):
...
+            seconds=constance.config.SYNCHRONOUS_EXPORT_CACHE_MAX_AGE
+        )
+        exports_with_same_settings = cls.objects.filter(
+            user=user,
+            asset_export_settings=asset_export_settings,
+            data__type=asset_export_settings.export_settings['type'],
+        )
+        # Clean up old exports regardless of status, assuming that synchronous
+        # exports aren't allowed to run longer than the maximum export age
+        exports_with_same_settings.filter(date_created__lt=age_cutoff).delete()
```
[the remark this answers is not in the document]
Comment (jnm) [reply kpi/models/import_export_task.py:850]: What would the advantage of that be?

File: kpi/models/import_export_task.py [hunk 2]
```diff
@@ -614,10 +800,84 @@ def remove_excess(cls, user, source):
...
+            .first()
+        )
+        if already_complete_export:
+            return already_complete_export
+        # Start a new export
+        data = asset_export_settings.export_settings.copy()
+        data['source'] = reverse(
+            'asset-detail', args=[asset_export_settings.asset.uid]
+        )
+        new_export = cls.objects.create(
```
[the remark this answers is not in the document]
Comment (jnm) [reply kpi/models/import_export_task.py:876]: Yes, I hope that the 5-minute caching plus the time limit on MongoDB queries will be enough without limiting to a small number of submissions.

File: kpi/models/import_export_task.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshuaberetta) [reply kpi/models/import_export_task.py:850]: Just thinking if someone doesn't make another export then outdated exports never get removed but I guess that's probably not a big deal.

File: kpi/models/import_export_task.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (joshuaberetta) [reply kpi/models/import_export_task.py:876]: Sounds good 👌

File: kpi/models/import_export_task.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jnm) [reply kpi/models/import_export_task.py]: My latest commit adds a guarantee that there'll never be more than one export per user+type+settings combination, but that export does stick around forever. I added #3718 for us to revisit this after the release.

**Review by joshuaberetta — APPROVED** [review]
✨

**Pull request merged**